services/scorecard/scorecard.service.js

```diff
@@ -0,0 +1,41 @@
+import Joi from 'joi'
+import { BaseJsonService } from '../index.js'
+
+const schema = Joi.object({
+  score: Joi.required(),
+}).required()
+
+export default class ScorecardScore extends BaseJsonService {
+  static category = 'analysis'
+
+  static route = { base: 'scorecard', pattern: ':host/:orgName/:repoName' }
+
+  async fetch({ host, orgName, repoName }) {
+    return this._requestJson({
+      schema,
+      url: `https://api.securityscorecards.dev/projects/${host}/${orgName}/${repoName}`,
+    })
+  }
+
+  async handle({ host, orgName, repoName }) {
+    const { score } = await this.fetch({ host, orgName, repoName })
+
+    let clr = ''
+
+    if (score <= 3) {
+      clr = 'red'
+    } else if (score <= 5) {
+      clr = 'orange'
+    } else if (score <= 7) {
+      clr = 'yellow'
+    } else {
+      clr = 'green'
+    }
```

Comment:
The scaled mapping of a numeric value to a range of colors is, as you might imagine, fairly common within Shields and we've a number of reusable helpers for such circumstances.

In this instance, I'd suggest taking a look at `colorScale` which can be imported from `color-formatters.js`, and [our code climate analysis class](https://github.com/badges/shields/blob/master/services/codeclimate/codeclimate-analysis.service.js) may serve as a helpful reference

Reply:
Fixed this using color-formatters.js, thanks for the reference

Reply:
Fixed this using color-formatters.js, thanks for the reference